extensions/logging-gelf/runtime/src/main/java/io/quarkus/logging/gelf/GelfConfig.java

```diff
@@ -61,4 +64,32 @@
      */
     @ConfigItem(defaultValue = "yyyy-MM-dd HH:mm:ss,SSS")
     public String timestampPattern;
+
+    /**
+     * The logging-gelf log level.
+     */
+    @ConfigItem(defaultValue = "ALL")
+    Level level;
+
+    /**
+     * Name of the Facility.
+     */
+    @ConfigItem(defaultValue = "jboss-logmanager")
+    String facility;
+
+    /**
+     * Post additional fields. Eg. `fieldName=Value,field2=value2`.
+     */
+    @ConfigItem
+    Optional<String> additionalFields;
+
+    /**
+     * Type specification for additional and MDC fields.
+     * Supported types: String, long, Long, double, Double and discover (default if not specified, discover field type on
+     * parseability).
+     * Eg. `field=String,field2=double`.
+     */
+    @ConfigItem
+    Optional<String> additionalFieldsTypes;
+
```

Comment:
Maybe this should be a map of config group?
 
Something like in pseudo code

```
@ConfigItem
Map<String, AdditionalField> additionalField;
```

And the `AdditionalField` class being 

```
@ConfigGroup
AdditionalField {
 @ConfigItem
 String value;

 @ConfigItem
 String type;
}

WDYT? @loicmathieu @gsmet

Reply:
I see your point, but I don't know if it will be easier to the user to configure them like this.

Current situation:
```
quarkus.log.handler.gelf.additional-fields=field1=value,field2=1
quarkus.log.handler.gelf.additional-fields-types=field1=String,field2=long
```

Your proposal:
```
quarkus.log.handler.gelf.additional-fields.field1.value=value
quarkus.log.handler.gelf.additional-fields.field1.type=String
quarkus.log.handler.gelf.additional-fields.field2.value=1
quarkus.log.handler.gelf.additional-fields.field2.type=long
```

Reply:
Thanks for the reply. 

With my proposal, the property will be in the generated doc. I think it also shows that the value and type go hand in hand (maybe that is pretty clear to some user but surely not all users). 

What happens if we provide fields value without the type or vice-versa with the current proposal?

Reply:
- field value without type: default type will be used (discover: auto mode based on parseability of the field)
- field type without value: no idea

Reply:
Great, thanks. 


> * field value without type: default type will be used (discover: auto mode based on parseability of the field)

Great thanks, just saw it on the Javadoc documentation. 

> * field type without value: no idea

Okay, can we have a test? If it is something not desirable, I think we should throw a `ConfigurationException`.  

Let us wait and see what others think of the two proposals or if we can find another option to make these configuration knobs more appealing to the user.